api/hypershift/v1beta1/azure_types.go

```diff
@@ -0,0 +1,213 @@
+
+	// diskStorageAccountType is the disk storage account type to use.
+	// Valid values are Standard, StandardSSD, PremiumSSD and UltraSSD and omitted.
+	// Note that Standard means a HDD.
+	// The disk performance is tied to the disk type, please refer to the Azure documentation for further details
+	// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-type-comparison.
+	// When omitted this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
+	// The current default is PremiumSSD.
+	//
+	// +kubebuilder:validation:Enum=Standard;StandardSSD;PremiumSSD;UltraSSD
+	// +optional
+	// TODO: Should all disk options come under a struct rather than be top level like this?
+	DiskStorageAccountType string `json:"diskStorageAccountType,omitempty"`
+
+	// availabilityZone is the failure domain identifier where the VM should be attached to.
+	// This must not be specified for clusters in a location that does not support AvailabilityZone because... TODO: why?
+	// Example values are TODO
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
+	//
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// diskEncryptionSetID is the ID of the DiskEncryptionSet resource to use to encrypt the OS disks for the VMs. This
+	// needs to exist in the same subscription id listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.SubscriptionID.
+	// DiskEncryptionSetID should also exist in a resource group under the same subscription id and the same location
+	// listed in the Hosted Cluster, HostedCluster.Spec.Platform.Azure.Location.
+	// TODO: What does this do for a customer, why would they want to set it?
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
+	//
+	// +optional
+	DiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
+
+	// TODO: This shouldn't be a bool, but I don't know what the field is really for.
+	// Should it be coming under the disk parameters? We need to know what the impact of this is for users.
+	// Is this a particularly bad idea for control plane nodes, and does that matter for HCP, probably not.
+	//
+	// enableEphemeralOSDisk is a flag when set to true, will enable ephemeral OS disk.
+	//
+	// +optional
+	EnableEphemeralOSDisk bool `json:"enableEphemeralOSDisk,omitempty"`
+
+	// subnetID is the subnet ID of an existing subnet where the nodes in the nodepool will be created. This can be a
+	// different subnet than the one listed in the HostedCluster, HostedCluster.Spec.Platform.Azure.SubnetID, but must
+	// exist in the same network, HostedCluster.Spec.Platform.Azure.VnetID, and must exist under the same subscription ID,
+	// HostedCluster.Spec.Platform.Azure.SubscriptionID.
+	// subnetID is immutable once set.
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
```

Comment:
For the actual name...

"The name must begin with a letter or number, end with a letter, number or underscore, and may contain only letters, numbers, underscores, periods, or hyphens.

The name must be between 1 and 80 characters."

Reply:
Is this a complete azure resource ID, or just the subnet itself?

Reply:
A complete azure resource ID